File: code/jvm/src/main/kotlin/pt/isel/gomoku/server/repository/dto/Stats.kt
package pt.isel.gomoku.server.repository.dto

data class WinStats(
    val totalWins: Int,
    val winsAsBlack: Int,
    val winsAsWhite: Int,
    val winRate: Double,
    val draws: Int,
    val loses: Int,
)

data class RawWinStats(
    val totalMatches: Int,
    val winsAsBlack: Int,
    val winsAsWhite: Int,
    val draws: Int
)

data class MatchesStats(
    val totalMatches: Int,
    val matchesAsBlack: Int,
    val matchesAsWhite: Int
)

File: code/js/src/types/stats.ts
export type UserStats = {
  rank: Rank;
  winStats: WinStats;
  matchesStats: MatchesStats;
};

export type Rank = {
  name: string;
  iconUrl: string;
};

export type WinStats = {
  totalWins: number;
  winsAsBlack: number;
  winsAsWhite: number;
  winRate: number;
  draws: number;
  loses: number;
};

export type MatchesStats = {
  totalMatches: number;
  matchesAsBlack: number;
  matchesAsWhite: number;
};

export type PaginationResult<T> = {
    results: T[];
    total: number;
}

export type UserItem = {
    id: number;
    name: string;
    role: string;
    count?: number | null;
    rank: {
        name: string;
        iconUrl: string;
    }
}